evm/src/fixed_recursive_verifier.rs

```diff
@@ -246,8 +326,54 @@ where
             );
             root_inputs.set_proof_with_pis_target(&self.root.proof_with_pis[table], &shrunk_proof);
         }
+
+        root_inputs.set_verifier_data_target(
+            &self.root.cyclic_vk,
+            &self.aggregation.circuit.verifier_only,
+        );
+
         self.root.circuit.prove(root_inputs)
     }
+
+    pub fn verify_root(&self, agg_proof: &ProofWithPublicInputs<F, C, D>) -> anyhow::Result<()> {
+        self.root.circuit.verify(agg_proof.clone())
+    }
```

Comment:
nit: Maybe take a value instead of a reference here to make the cloning explicit.

Reply:
Done. I was thinking it might be good to have `verify` itself take a reference, since it doesn't really take ownership of any data, but we can consider it later.